File: mirai-core/src/main/java/net/mamoe/mirai/utils/ImageNetworkUtils.java
package net.mamoe.mirai.utils;

import org.apache.commons.httpclient.util.HttpURLConnection;

import java.io.IOException;
import java.net.URL;

/**
 * @author NaturalHG
 */
public class ImageNetworkUtils {
    public static void postImage(String uKeyHex, int fileSize, String qqNumber, String groupCode, byte[] img) throws IOException {
        //http://htdata2.qq.com/cgi-bin/httpconn?htcmd=0x6ff0071&ver=5515&term=pc&ukey=” ＋ 删全部空 (ukey) ＋ “&filesize=” ＋ 到文本 (fileSize) ＋ “&range=0&uin=” ＋ g_uin ＋ “&groupcode=” ＋ Group
        StringBuilder builder = new StringBuilder("http://htdata2.qq.com/cgi-bin/httpconn?htcmd=0x6ff0071&ver=5515&term=pc");
        builder.append("&ukey=")
                .append(uKeyHex.trim())
                .append("&filezise=").append(fileSize)
                .append("&range=").append("0")
                .append("&uin=").append(qqNumber)
                .append("&groupcode=").append(groupCode);

        HttpURLConnection conn = (HttpURLConnection) new URL(builder.toString()).openConnection();
        conn.setRequestProperty("User-agent", "QQClient");
        conn.setRequestProperty("Content-length", "" + fileSize);
        conn.setRequestMethod("POST");
        conn.getOutputStream().write(img);

        conn.connect();
        System.out.println(conn.getResponseCode());
        System.out.println(conn.getResponseMessage());
    }
}

File: mirai-core/src/jvmMain/kotlin/net/mamoe/mirai/utils/ImageNetworkUtils.kt
package net.mamoe.mirai.utils

import java.io.IOException
import java.net.HttpURLConnection
import java.net.URL

/**
 * @author NaturalHG
 */
object ImageNetworkUtils {
    @Throws(IOException::class)
    fun postImage(uKeyHex: String, fileSize: Int, botNumber: Long, groupCode: Long, img: ByteArray): Boolean {
        //http://htdata2.qq.com/cgi-bin/httpconn?htcmd=0x6ff0071&ver=5515&term=pc&ukey=” ＋ 删全部空 (ukey) ＋ “&filesize=” ＋ 到文本 (fileSize) ＋ “&range=0&uin=” ＋ g_uin ＋ “&groupcode=” ＋ Group

        val builder = "http://htdata2.qq.com/cgi-bin/httpconn?htcmd=0x6ff0071&ver=5515&term=pc" +
                "&ukey=" + uKeyHex.replace(" ", "") +
                "&filezise=" + fileSize +
                "&range=" + "0" +
                "&uin=" + botNumber +
                "&groupcode=" + groupCode
        val conn = URL(builder).openConnection() as HttpURLConnection
        conn.setRequestProperty("User-Agent", "QQClient")
        conn.setRequestProperty("Content-Length", "" + fileSize)
        conn.requestMethod = "POST"
        conn.doOutput = true
        conn.outputStream.write(img)

        conn.connect()
        return conn.responseCode == 200
    }
}
